src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/client/ManageUsersApiClient.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.client
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+
+@Component
+class ManageUsersApiClient(
+  @Qualifier("manageUsersApiWebClient") private val webClient: WebClient,
+) {
+  fun getUsername() = webClient.get()
+    .uri("/users/me")
+    .accept(MediaType.APPLICATION_JSON)
+    .retrieve()
+    .bodyToMono<UserNameDetails>()
+    .block()!!
+
+  fun getUserDetails(userName: String) =
+    webClient.get()
+      .uri("/users/$userName")
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<UserDetails>()
+      .block()!!
+}
+
+data class UserDetails(val activeCaseLoadId: String? = null)
+data class UserNameDetails(val username: String? = null)
```

Comment:
I would put these above ManageUsersApiClient

Reply:
thanks. The original code has it below the Client calls - so will not update it - https://github.com/ministryofjustice/send-legal-mail-to-prisons-api/pull/221/files#diff-745c2f8d0a37b568421f555c18879e98b230b6ae4107d31aab9682d8598a6427